### ISSUE TITLE
Automatically find referenced fonts

### ISSUE THREAD

**Issue by ianthetechie** [stadiamaps/cartogrify#3]
Currently, cartogrify does not generate a `fonts` section. However, with a bit of work, we could probably look up many fonts using the Google Fonts API.

First we need to investigate whether this is possible. If yes, we can implement it with the API key as a command line switch. A bit of parsing to separate font families and names out may also be necessary in the process.

**Comment by nvkelso** [comment]
I think Tangram supports CSS-style font fallback with comma-delimited lists?

I don't see it documented, though: 

- https://tangrams.readthedocs.io/en/master/Syntax-Reference/draw/#font-parameters

**Comment by ianthetechie** [comment]
@nvkelso thanks for the tip! I'll give that at try and can submit a PR for the Tangram docs if that works.

**Comment by ianthetechie** [comment]
Thanks for pointing this out! It seems to me like fallback does work, even though it's not documented. @bcamper you seem to have done a fair bit with the font related code on Tangram... Is it as simple as it sounds (comma separated values dictate a CSS-style fallback order)? Or is there anything else that should be documented here?

In tangentially related news, I ran into some issues with getting it to recognize externally loaded fonts and opened an issue (https://github.com/tangrams/tangram/issues/771). Do either of you guys know of any pros/cons when it comes to CSS-loaded vs fonts specified directly in the scene? It looks like CSS fonts may not work in non-web contexts (ex: tangram-es) so I think the Google Font API idea is probably still worth pursuing.

**Comment by bcamper** [comment]
Yes, the font fallback is indeed just a passthrough to CSS-style font fallbacks, because the text is being rendered using Canvas (to a WebGL texture). So yes it should be documented with a "JS only" note.

And yes, due to significant differences in the text rendering pipeline between Tangram JS (uses Canvas text rendering) and ES (uses SDF text rendering, similar to Mapbox but with SDF generation done client-side to support any OpenType font, plus text shaping for Arabic + Indic fonts, etc.), both the `external` and CSS-comma-fallback syntax are currently _only_ supported in Tangram JS.

So the best approach for maximum compatibility is to explicitly define each font face you need in `fonts`, and I have loaded fonts from Google Fonts API like that with good results.
